### PR TITLE
Simplify Table rendering. Move allocations out of Cell-functions, avoiding unnecessary allocations

### PR DESCRIPTION
**Pull request by awildturtok**
Die Methoden auszulagern fehlt mir die Zeit aktuell, würde den Code aber deutlich lesbarer machen.
Ich werde demnächst einen PR stellen, indem ich das mit der currency-Code richtig implementiere, dann ist das auch sauberer.

### CODE REVIEW

File: frontend/src/js/preview/tableUtils.ts [hunk 1]
```diff
@@ -20,7 +15,18 @@ export function useCustomTableRenderers(queryData: GetQueryResponseDoneT) {
...
+      const dateFormatter = new Intl.DateTimeFormat(navigator.language, {
+        day: "2-digit",
+        month: "2-digit",
+        year: "numeric",
+      });
+
+      const currencyFormatter = new Intl.NumberFormat(navigator.language, {
+        style: "currency",
+        currency: currencyFromSymbol(currencyConfig.unit),
+      });
```
Comment (Kadrian) [inline frontend/src/js/preview/tableUtils.ts:28]: How about extracting these out, so we don't re-initialize the formatters on every cell render?

Comment (awildturtok) [reply frontend/src/js/preview/tableUtils.ts:28]: The changes in Table.tsx#72 should ensure, that this method is called only once per column.

Comment (Kadrian) [reply frontend/src/js/preview/tableUtils.ts:28]: Aah, I see 👍

File: frontend/src/js/preview/tableUtils.ts [hunk 2]
```diff
@@ -20,7 +15,18 @@ export function useCustomTableRenderers(queryData: GetQueryResponseDoneT) {
   );
 
   const getRenderFunction = useCallback(
-    (cellType: string): ((value: CellValue) => string) | undefined => {
+    (cellType: string): ((value: CellValue) => string) => {
+      const dateFormatter = new Intl.DateTimeFormat(navigator.language, {
```
Comment (Kadrian) [inline frontend/src/js/preview/tableUtils.ts:19]: Good idea actually to use this new API instead of our usual `dateHelper` functions.

**Review by Kadrian — APPROVED** [review]
Looks good to me! 👍 Good idea to limit to 100 lines for now and to avoid manually attaching window scroll listeners and doing scroll pos calculations

**Pull request merged**